### PR TITLE
Email users when a document of a particular schema is created if configured

### PR DESCRIPTION
**Pull request by kunwarj**

### CODE REVIEW

File: nunaliit2-couch-submission/src/main/java/ca/carleton/gcrc/couch/submission/mail/DocumentCreatedGenerator.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+		} else {
+			message.setSubject("Document created - "+docId);
+		}
+		
+		// Create HTML body part
+		StringWriter sw = new StringWriter();
+		PrintWriter pw = new PrintWriter(sw);
+		pw.println("<html><head><title>Document Created</title></head><body><h1>Document Created</h1>");
+		
+        pw.println("<p>A new document has been created for schema: "+schemaName+".</p>");
```
Comment (alexgao1) [inline nunaliit2-couch-submission/src/main/java/ca/carleton/gcrc/couch/submission/mail/DocumentCreatedGenerator.java:33]: Formatting

File: nunaliit2-couch-submission/src/main/java/ca/carleton/gcrc/couch/submission/mail/SubmissionMailNotifierImpl.java [hunk 2]
```diff
@@ -224,4 +227,75 @@ public void sendSubmissionRejectionNotification(
...
+			if (userIds.add(user.getId())) {
+				String display = user.getDisplayName();
+				for (String email : user.getEmails()) {
+					recipients.add(display == null ? new MailRecipient(email) : new MailRecipient(email, display));
+				}
+			}
+        }
+
+		if (recipients.isEmpty()) {
+            logger.info("Document created notification not sent because there are no recipients");
```
Comment (alexgao1) [inline nunaliit2-couch-submission/src/main/java/ca/carleton/gcrc/couch/submission/mail/SubmissionMailNotifierImpl.java:264]: Formatting

File: nunaliit2-couch-submission/src/main/java/ca/carleton/gcrc/couch/submission/mail/SubmissionMailNotifierImpl.java [hunk 3]
```diff
@@ -224,4 +227,75 @@ public void sendSubmissionRejectionNotification(
...
+            return;
+        }
+
+		logger.info("Sending documnet created mail notification for "
```
Comment (alexgao1) [inline nunaliit2-couch-submission/src/main/java/ca/carleton/gcrc/couch/submission/mail/SubmissionMailNotifierImpl.java:268]: Typo

**Review by alexgao1 — CHANGES_REQUESTED** [review]
Notes so far: Seems to only email vetters, doesn't seem to email atlas adminstrators.
I think that despite the fact that it would email twice, when the submission server is on, it should send a "document was created for the schema" email plus the typical "a submission was made" email.

File: nunaliit2-couch-submission/src/main/java/ca/carleton/gcrc/couch/submission/mail/SubmissionMailNotifierImpl.java [hunk 4]
```diff
@@ -224,4 +227,75 @@ public void sendSubmissionRejectionNotification(
...
+
+		if( null != currentUser ) {
+			users.add(currentUser);
+		}
+
+		List<String> roles = new ArrayList<String>(2);
+		roles.add("vetter"); // global vetters
+		roles.add(atlasName+"_vetter"); // atlas vetters
+		roles.add("administrator"); // global administrator
+		roles.add(atlasName+"administrator"); // atlas administrator
```
Comment (alexgao1) [inline nunaliit2-couch-submission/src/main/java/ca/carleton/gcrc/couch/submission/mail/SubmissionMailNotifierImpl.java:246]: Typo, missing "_"

**Comment by alexgao1** [comment]
It should also log that an email cannot send if there is no mail server

File: nunaliit2-mail/src/main/java/ca/carleton/gcrc/mail/MailDeliveryImpl.java [hunk 5]
```diff
@@ -32,7 +32,15 @@ public MailDeliveryImpl() {
 
 	@Override
 	public boolean isConfigured() {
-		if( null == fromAddress ) {
+
+		String prot = mailProperties.getProperty("mail.transport.protocol",null);
+		String userName = mailProperties.getProperty("mail."+prot+".user",null);
```
Comment (alexgao1) [inline nunaliit2-mail/src/main/java/ca/carleton/gcrc/mail/MailDeliveryImpl.java:37]: Unsure if mail. ... .user and .password are used as a part of mail configuration normally

**Review by alexgao1 — CHANGES_REQUESTED** [review]

File: nunaliit2-couch-submission/src/main/java/ca/carleton/gcrc/couch/submission/mail/SubmissionMailNotifierImpl.java [hunk 6]
```diff
@@ -224,4 +227,81 @@ public void sendSubmissionRejectionNotification(
...
 	}
 
+	@Override
+	public void sendDocumentCreatedNotification(
+		JSONObject submissionDoc,
+		UserDocument currentUser
+		) throws Exception {
+
+		if(!mailDelivery.isConfigured()) {
+			logger.info("Email server is not cofigured to send document created notification");
```
Comment (alexgao1) [inline nunaliit2-couch-submission/src/main/java/ca/carleton/gcrc/couch/submission/mail/SubmissionMailNotifierImpl.java:237]: typo

File: nunaliit2-couch-submission/src/main/java/ca/carleton/gcrc/couch/submission/mail/SubmissionMailNotifierImpl.java [hunk 7]
```diff
@@ -224,4 +227,81 @@ public void sendSubmissionRejectionNotification(
...
+		roles.add(atlasName+"_vetter"); // atlas vetters
+		roles.add("administrator"); // global administrator
+		roles.add(atlasName+"_administrator"); // atlas administrator
+
+		users.addAll(new ArrayList<>(userDesignDocument.getUsersWithRoles(roles)));
+
+		List<MailRecipient> recipients = new ArrayList<>();
+
+		// create a unique set to userIds, so as we don't send duplicate mail if the current user has admin, or vetter roles
+		Set<String> userIds = new HashSet<>();
```
Comment (kunwarj) [inline nunaliit2-couch-submission/src/main/java/ca/carleton/gcrc/couch/submission/mail/SubmissionMailNotifierImpl.java:258]: Set and HasSet are being used here to create a unique set of userId's

**Review by alexgao1 — APPROVED** [review]

**Pull request merged**